tests/test_frame_db.py

```diff
@@ -38,3 +57,191 @@ def test_fast_query_warning(test_db):
         FrameIndex.open("sqlite:///" + test_db, mode="r")
 
     return
+
+
+def test_year_month_str_key():
+    t_start = Time("2006-01-02 15:04:05")
+    t_mid = Time("2007-02-02 15:04:05")
+
+    frame = SourceFrame(
+        exposure_id="id",
+        obscode="n/a",
+        filter="n/a",
+        exposure_mjd_start=t_start.mjd,
+        exposure_mjd_mid=t_mid.mjd,
+        exposure_duration=1,
+        healpixel=1,
+        observations=[],
+    )
+    assert FrameDB._compute_year_month_str(frame) == "2007-02"
+
+
+def test_add_dataset_creates_if_missing(frame_db):
```

Comment:
Beyond scope, but since this is supposed to be idempotent and we support multiprocessing it might be good to test that the upsert / locking functionality is working.

Reply:
Multiprocessing tests are something I have not done much. I agree it's worth doing.